### PR TITLE
Ps config field converter

### PR DESCRIPTION
**Pull request by psantos1113**
# Pull Request template

**Changes proposed in this pull request:**

* Convert FieldModel to the correct ConfigurationFieldModel
* Fix the saving of the settings to preserve a single configuration with proper converted fields
*

### CODE REVIEW

File: src/main/java/com/synopsys/integration/alert/common/descriptor/Descriptor.java [hunk 1]
```diff
@@ -129,14 +130,20 @@ public boolean hasUIConfigForType(final ConfigContextEnum actionApiType) {
...
-                               actionApi.ifPresent(descriptorActionApi -> descriptorActionApi.validateConfig(config.createFields(), fieldModel, fieldErrors)));
+                               actionApi.ifPresent(descriptorActionApi -> descriptorActionApi.validateConfig(configFieldMap, fieldModel, fieldErrors)));
     }
 
     public void testConfig(final ConfigContextEnum actionApiType, final TestConfigModel testConfig) throws IntegrationException {
         final Optional<DescriptorActionApi> actionApi = getActionApi(actionApiType);
         if (actionApi.isPresent()) {
-            final List<ConfigField> configFields = retrieveUIConfigFields(actionApiType, testConfig.getFieldModel());
+            final Map<String, ConfigField> configFields = createConfigFieldMap(actionApiType);
+            ;
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/common/descriptor/Descriptor.java]: I think this was meant to be deleted

File: src/main/java/com/synopsys/integration/alert/web/config/ConfigActions.java [hunk 2]
```diff
@@ -165,8 +179,20 @@ public String testConfig(final FieldModel restModel, final String destination) t
...
+                                                              .collect(Collectors.toMap(ConfigField::getKey, Function.identity()));
+            final ConfigContextEnum descriptorContext = EnumUtils.getEnum(ConfigContextEnum.class, restModel.getContext());
+
+            if (ConfigContextEnum.DISTRIBUTION == descriptorContext) {
+                final Map<String, ConfigField> globalFields = createConfigFields(descriptorContext, restModel.getDescriptorName()).stream()
+                                                                  .collect(Collectors.toMap(ConfigField::getKey, Function.identity()));
+                fieldsToTest.putAll(globalFields);
+                fieldsToTest.putAll(includeProviderFields(descriptorContext, restModel));
+            }
+            fieldsToTest.putAll(configFields);
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/web/config/ConfigActions.java]: We should convert this map to FieldAccessor before passing it to the actionApi. All the child classes that define this method immediately convert it into FieldAccessor.

File: src/main/java/com/synopsys/integration/alert/web/controller/metadata/DescriptorController.java [hunk 3]
```diff
@@ -95,10 +101,31 @@ public DescriptorController(final Collection<Descriptor> descriptors) {
...
         for (final ConfigContextEnum applicableContext : applicableContexts) {
             for (final Descriptor descriptor : filteredDescriptors) {
-                final Optional<DescriptorMetadata> uiConfig = descriptor.getMetaData(applicableContext);
-                uiConfig.ifPresent(descriptorMetadata::add);
+                final Optional<DescriptorMetadata> optionalMetaData = descriptor.getMetaData(applicableContext);
+                if (ConfigContextEnum.DISTRIBUTION == applicableContext) {
+                    optionalMetaData.ifPresent(metadata -> descriptorMetadata.add(createMetaData(metadata)));
+                } else {
+                    optionalMetaData.ifPresent(descriptorMetadata::add);
+
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/web/controller/metadata/DescriptorController.java]: Spacing :D

File: src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionUIConfig.java [hunk 4]
```diff
@@ -42,8 +42,9 @@ public EmailDistributionUIConfig() {
 
     @Override
     public List<ConfigField> createFields() {
+        final ConfigField emailAddresses = TextInputConfigField.create(EmailDescriptor.KEY_EMAIL_ADDRESSES, "Email Addresses");
```
[the remark this answers is not in the document]
Comment (psantos1113) [reply src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionUIConfig.java]: yes the UI does not render these fields and will not in the next release.  A future release will.

File: src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailGlobalDescriptorActionApi.java [hunk 5]
```diff
@@ -71,7 +74,7 @@ public void testConfig(final Collection<ConfigField> configFields, final TestCon
             }
             emailAddresses = Set.of(testEmailAddress);
         }
-        final FieldAccessor fieldAccessor = testConfig.getFieldModel().convertToFieldAccessor();
+        final FieldAccessor fieldAccessor = modelConverter.convertToFieldAccessor(configFields, testConfig.getFieldModel());
```
[the remark this answers is not in the document]
Comment (psantos1113) [reply src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailGlobalDescriptorActionApi.java]: I'll look into it

**Pull request merged**